scipy/spatial/transform/rotation.py

```diff
@@ -0,0 +1,754 @@
+        .. [3] `Gimbal lock
+                <https://en.wikipedia.org/wiki/Gimbal_lock#In_applied_mathematics>`_
+        """
+        if len(seq) != 3:
+            raise ValueError("Expected 3 axes, got {}.".format(seq))
+
+        intrinsic = (re.compile('^[XYZ]{1,3}$').match(seq) is not None)
+        extrinsic = (re.compile('^[xyz]{1,3}$').match(seq) is not None)
+        if not (intrinsic or extrinsic):
+            raise ValueError("Expected axes from `seq` to be from "
+                             "['x', 'y', 'z'] or ['X', 'Y', 'Z'], "
+                             "got {}".format(seq))
+
+        if any(seq[i] == seq[i+1] for i in range(2)):
+            raise ValueError("Expected consecutive axes to be different, "
+                             "got {}".format(seq))
+
+        seq = seq.lower()
+
+        angles = _compute_euler_from_dcm(self.as_dcm(), seq, extrinsic)
+        if degrees:
+            angles = np.rad2deg(angles)
+
+        return angles[0] if self._single else angles
+
+    def inv(self):
+        """Returns the inverse of the current rotation.
+
+        This function returns a new `Rotation` instance containing the inverse
+        of all the rotations in the current instance.
+        """
+        quat = self._quat.copy()
+        quat[:, -1] *= -1
+        if self._single:
+            quat = quat[0]
+        return self.__class__(quat, normalized=True, copy=False)
+
+    def __mul__(self, other):
+        """Compose this rotation with the other.
+
+        If `p` and `q` are two rotations, then the composition of 'q followed
+        by p' is equivalent to `p * q`. In terms of DCMs, the composition can
+        be expressed as `p.as_dcm().dot(q.as_dcm())`.
+
+        This function supports composition of multiple rotations at a time:
+
+            - Either `p` or `q` contains a single rotation. In this case the
+              returned object contains the result of composing each rotation in
+              the other object with the single rotation.
+            - Both `p` and `q` contain `N` rotations. In this case each
+              rotation `p[i]` is composed with each rotation `q[i]` and the
+              returned object contains `N` rotations.
+        """
+        if not(self._quat.shape[0] == 1 or other._quat.shape[0] == 1 or
+               self._quat.shape[0] == other._quat.shape[0]):
+            raise ValueError("Expected equal number of rotations in both "
+                             "or a single rotation in either object, "
+                             "got {} rotations in first and {} rotations in "
+                             "second object.".format(
+                                self._quat.shape[0], other._quat.shape[0]))
+        result = _compose_quat(self._quat, other._quat)
+        if self._single and other._single:
+            result = result[0]
+        return self.__class__(result, normalized=True, copy=False)
+
+    def apply(self, vectors, inverse=False):
```

Comment:
I'm not sure `inverse` is needed. One can just as easily do `rots.inv().apply(vecs)` if they want this behavior

Reply:
How do we resolve this suggestions? I'm slightly in favor for keeping `inverse`, because this is how I'm used to think about rotation application --- one objects does rotations in both directions. In other words, we don't need to have a rotation and its inverse to do the job.